### PR TITLE
add dependency list to docs

### PR DESCRIPTION
**Pull request by esc**
as title

### CODE REVIEW

File: docs/source/user/installing.rst [hunk 1]
```diff
@@ -176,6 +176,54 @@ Then you can build and install Numba from the top level of the source tree::
...
+  * ``scipy`` - provides cython bindings used in Numba's ``np.linalg.*`` support
+  * ``tbb`` - provides the TBB runtime libraries used by Numba's TBB threading backend
+  * ``jinja2`` - for "pretty" type annotation output (HTML) via the ``numba`` CLI
+  * ``cffi`` - permits use of CFFI bindings in Numba compiled functions
+  * ``intel-openmp`` - (OSX) provides OpenMP library support for Numba's OpenMP threading backend
+  * ``ipython`` - if in use, caching will use IPython's cache directories/caching still works
+  * ``pyyaml`` - permits the use of a ``.numba_config.yaml`` file for storing per project configuration options
+  * ``colorama`` - makes error message highlighting work
+  * ``icc_rt`` - (numba channel) allows Numba to use Intel SVML for extra performance
+  * ``pygments`` - for "pretty" type annotation
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Also compilers can go here for `pycc` AOT compilation.

Comment (stuartarchibald) [reply docs/source/user/installing.rst]: Not sure if we should mention gdb here too?

File: docs/source/user/installing.rst [hunk 2]
```diff
@@ -176,6 +176,54 @@ Then you can build and install Numba from the top level of the source tree::
...
 .. _numba-source-install-check:
 
+Dependency List
+---------------
+
+Numba has numerous required and optional dependencies which additionally may
+vary with target operating system and hardware. The following lists them all
+(as of September 2019).
+
+* Required build time:
```
[the remark this answers is not in the document]
Comment (esc) [reply docs/source/user/installing.rst]: Maybe we could list examples? Like the conda packages for linux and osx? (Which I still don't know what they are).

[the remark this answers is not in the document]
Comment (esc) [reply docs/source/user/installing.rst]: Is it just `clang`?

File: docs/source/user/installing.rst [hunk 3]
```diff
@@ -176,6 +176,54 @@ Then you can build and install Numba from the top level of the source tree::
...
+
+  * ``setuptools``
+  * ``numpy``
+  * ``llvmlite``
+  * ``funcsigs`` (Python 2)
+  * ``singledispatch`` (Python 2)
```
[the remark this answers is not in the document]
Comment (esc) [reply docs/source/user/installing.rst]: Yeah, I don't think there is much point in adding 3.4 deps at this stage.

File: docs/source/user/installing.rst (hunk 1 above)
Comment (esc) [reply docs/source/user/installing.rst]: So, just `pycc` and `gdb`?

File: docs/source/user/installing.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply docs/source/user/installing.rst]: For conda packages...

On linux this:
* `gcc_linux-64` for `x86_64`
* `gcc_linux-32` for `x86`
* `gcc_linux-ppc64le` for POWER
* There are no packages for ARM on the Anaconda distribution channels so the system compiler is used.

On OSX:
* `clang_osx-64`

The C++ compilers are the same but s/clang/clangxx/ and s/gcc/gxx/

File: docs/source/user/installing.rst (hunk 1 above)
Comment (stuartarchibald) [reply docs/source/user/installing.rst]: `pycc` is the Numba tool for performing AOT compilation, it relies on a working tool chain, e.g. https://github.com/numba/numba/pull/4607#discussion_r327992817. The `gdb` support relies on a `gdb` binary being present on `$PATH`.

File: docs/source/user/installing.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (esc) [reply docs/source/user/installing.rst]: As for OSX - I use the system compiler at `/usr/bin/clang`. I'll mention that in the docs.

File: docs/source/user/installing.rst [hunk 4]
```diff
@@ -165,9 +165,15 @@ Source archives of the latest release can also be found on
 `PyPI <https://pypi.org/project/numba/>`_.  In addition to ``llvmlite``, you will also need:
 
 * A C compiler compatible with your Python installation.  If you are using
-  Anaconda, you can install the Linux compiler conda packages ``gcc_linux-64``
-  and ``gxx_linux-64``, or macOS packages ``clang_osx-64`` and
-  ``clangxx_osx-64``.
+  Anaconda, you can use the follwing conda packages:
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]:
```suggestion
  Anaconda, you can use the following conda packages:
```

File: docs/source/user/installing.rst [hunk 5]
```diff
@@ -165,9 +165,15 @@ Source archives of the latest release can also be found on
...
+
+  * Linux ``x86``: ``gcc_linux-32`` and ``gxx_linux-32``
+  * Linux ``x86_64``: ``gcc_linux-64`` and ``gxx_linux-64``
+  * Linux ``POWER``: ``gcc_linux-ppc64le``
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: The C++ compiler `gxx_linux-ppc64le` is also needed.

File: docs/source/user/installing.rst [hunk 6]
```diff
@@ -165,9 +165,15 @@ Source archives of the latest release can also be found on
...
+  * Linux ``ARM``: no conda packages, use the system compiler
+  * Mac OSX: ``clang_osx-64`` and ``clangxx_osx-64`` or the sytem compiler at
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]:
```suggestion
  * Mac OSX: ``clang_osx-64`` and ``clangxx_osx-64`` or the system compiler at
```

File: docs/source/user/installing.rst [hunk 7]
```diff
@@ -165,9 +165,15 @@ Source archives of the latest release can also be found on
...
+    ``/usr/bin/clang`` (Mojave onwards).
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Probably need to say something about visual studio here too?

File: docs/source/user/installing.rst [hunk 8]
```diff
@@ -176,6 +182,68 @@ Then you can build and install Numba from the top level of the source tree::
...
+    threading backend
+  * ``ipython`` - if in use, caching will use IPython's cache
+    directories/caching still works
+  * ``pyyaml`` - permits the use of a ``.numba_config.yaml``
+    file for storing per project configuration options
+  * ``colorama`` - makes error message highlighting work
+  * ``icc_rt`` - (numba channel) allows Numba to use Intel SVML for extra
+    performance
+  * ``pygments`` - for "pretty" type annotation
+  * ``gdb`` - if you would like to us ethe gdb support
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]:
```suggestion
  * ``gdb`` - if you would like to use the gdb support
```
also probably work making clear that this is a system binary not a package?

File: docs/source/user/installing.rst (hunk 5 above)
Comment (esc) [reply docs/source/user/installing.rst]: Yup

File: docs/source/user/installing.rst (hunk 7 above)
Comment (esc) [reply docs/source/user/installing.rst]: Yup

File: docs/source/user/installing.rst (hunk 1 above)
Comment (esc) [reply docs/source/user/installing.rst]: Yup

**Comment by esc** [comment]
Suggestions have (hopefully) all been addressed.

File: docs/source/user/installing.rst [hunk 9]
```diff
@@ -165,9 +165,16 @@ Source archives of the latest release can also be found on
...
-  ``clangxx_osx-64``.
+  Anaconda, you can use the following conda packages:
+
+  * Linux ``x86``: ``gcc_linux-32`` and ``gxx_linux-32``
+  * Linux ``x86_64``: ``gcc_linux-64`` and ``gxx_linux-64``
+  * Linux ``POWER``: ``gcc_linux-ppc64le`` and ``gxx_linux-ppc64le``
+  * Linux ``ARM``: no conda packages, use the system compiler
+  * Mac OSX: ``clang_osx-64`` and ``clangxx_osx-64`` or the system compiler at
+    ``/usr/bin/clang`` (Mojave onwards).
+  * Windows: an approprite version of Visual Studio that can compile Python
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Perhaps `an version of Visual Studio appropriate for the Python version in use` ?

File: docs/source/user/installing.rst [hunk 10]
```diff
@@ -176,6 +183,69 @@ Then you can build and install Numba from the top level of the source tree::
...
+    support
+  * Compiler toolchain mentioned above, if you would like to use ``pycc`` for
+    Ahead-of-Time (AOT) compilation
+
+* To build the documentation:
+
+  * ``sphinx``
+  * ``pygments``
+  * ``sphinx-bootstrap``
+  * ``numpydoc``
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: I think this technically also requires a working ``make``.

Comment (esc) [reply docs/source/user/installing.rst]: correct

File: docs/source/user/installing.rst (hunk 9 above)
Comment (esc) [reply docs/source/user/installing.rst]: Sure

**Comment by esc** [comment]
Alright, changes commited and pushed.

**Pull request merged**